src/card.rs

```diff
@@ -1,22 +1,366 @@
-use std::str::Utf8Error;
+use std::convert::TryFrom;
 
 use crate::error::Error;
+/// Holds eight bytes of ASCII characters, i.e. the length of a FITS compliant keyword.
 pub type Keyword = [u8; 8];
+/// Holds 80 bytes of ASCII characters, i.e. one line in a FITS compliant file.
+pub type CardBuf = [u8; 80];
 
 use serde::Serialize;
-#[derive(Debug, PartialEq, Serialize)]
-pub struct Card {
-    pub kw: Keyword,
-    pub v: Value,
+
+/// Enum representing the variants of a single 80 character line in the header.
+#[derive(PartialEq, Debug, Serialize)]
+pub enum Card {
+    /// A keyword card with a value and optional comment.
+    Value {
+        name: String,
+        value: Value,
+    },
+    /// Continuation of a long-string with CONTINUE in the keyword field of the card.
+    Continuation {
+        string: Option<String>,
+        comment: Option<String>,
+    },
+    // FITS extension string value, cf FITSv4, section 4.2.1.1
+    Extension([u8; 8]),
```

Comment:
Should put an enum here Extension that could have only 3 possibilities, Image, Table and BinaryTable ?

Reply:
Yes, especially since there is already an enum defined in `fitsrs::hdu::header::extension`:
```rust
pub enum XtensionType {
    Image,
    BinTable,
    AsciiTable,
}
```

Reply:
Done.